### DISCUSSION TITLE
Newcomer's request for help, turn on LED

### DISCUSSION THREAD

**Discussion by carl336** [micropython/micropython#13652] [RP2040 / Pico]
Hi, I'm using Raspberry Pi Pico with Micropython, here is the program code I wrote:
 https://paste.fastmirror.net/?a598cbd31646761f#5EPdjtowXjxu2hAjcCviDUkHU3KVzZc6giyLcAa7WAtw 

I would like to ask you to teach me how to modify the code to follow the following process：
1. When I press KEY1 for the 1st time,  
gs_data_IC_1_1()～gs_data_IC_1_17() executed, LED pattern 1 turned on.
2. When I press the KEY2 for the 2nd time,
 gs_data_IC_2_1()～gs_data_IC_2_17() executed, LED pattern 2 turned on.
3. When I press the KEY2 for the 3rd time, 
gs_data_off executed, all LEDs turned off.
4. When I press KEY1 for the 1st time,  
gs_data_IC_1_1()～gs_data_IC_1_17() executed, LED pattern 1 turned on......
5. Enter an infinite loop......

**Comment by scruss** [comment]
Your code is impossible to follow. I think you need to learn to program from the beginning.

**Comment by sosi-deadeye** [comment]
The data which is sent via SPI to control the LEDs could be calculated or saved on flash as bytes. Putting everything into the source code, makes the program unflexible.
